src/main/scala/purecsv/safe/package.scala

```diff
@@ -41,49 +41,56 @@ package object safe {
-
+  implicit def optionc[A](implicit ac: StringConverter[A]): StringConverter[Option[A]] =
+    purecsv.safe.converter.defaults.string.optionc
 
   // Raw Fields
   implicit val deriveHNil: RawFieldsConverter[HNil] = purecsv.safe.converter.defaults.rawfields.deriveHNil
 
   implicit def deriveHCons[V, T <: HList](implicit sc: StringConverter[V],
-                                          fto: RawFieldsConverter[T])
-  : RawFieldsConverter[V :: T] = {
+                                          fto: RawFieldsConverter[T]): RawFieldsConverter[V :: T] = {
     purecsv.safe.converter.defaults.rawfields.deriveHCons
   }
 
   implicit def deriveClass[A, R](implicit gen: Generic.Aux[A, R],
-                                 conv: RawFieldsConverter[R])
-  : RawFieldsConverter[A] = {
+                                 conv: RawFieldsConverter[R]): RawFieldsConverter[A] = {
     purecsv.safe.converter.defaults.rawfields.deriveClass
   }
 
   implicit class CSVRecord[A](a: A)(implicit rfc: RawFieldsConverter[A])
-    extends purecsv.csviterable.CSVRecord[A, RawFieldsConverter[A]](a)(rfc)
+      extends purecsv.csviterable.CSVRecord[A, RawFieldsConverter[A]](a)(rfc)
 
   implicit class CSVIterable[A](iter: Iterable[A])(implicit rfc: RawFieldsConverter[A])
-    extends purecsv.csviterable.CSVIterable[A, RawFieldsConverter[A]](iter)(rfc)
+      extends purecsv.csviterable.CSVIterable[A, RawFieldsConverter[A]](iter)(rfc)
 
   trait CSVReader[A] extends Serializable {
 
     def rfc: RawFieldsConverter[A]
 
-    def readCSVFromReader(r: Reader,
-                          delimiter: Char = RecordSplitter.defaultFieldSeparator,
-                          trimming: Trimming = NoAction,
-                          headers: Headers = ParseHeaders)(implicit typeTag: TypeTag[A]): Iterator[Try[A]] = {
-      RecordSplitterImpl.getRecords(r, caseClassParams[A], delimiter, trimming = trimming, headers = headers)
+    def readCSVFromReader(
+        r: Reader,
+        delimiter: Char = RecordSplitter.defaultFieldSeparator,
+        trimming: Trimming = NoAction,
+        headers: Headers = ParseHeaders,
+        headerMapping: Map[String, String] = Map())(implicit typeTag: TypeTag[A]): Iterator[Try[A]] = {
+      RecordSplitterImpl
+        .getRecords(r,
+                    caseClassParams[A],
+                    delimiter,
+                    trimming = trimming,
+                    headers = headers,
+                    headerMapping = headerMapping)
         .map(record => rfc.tryFrom(record.toSeq))
     }
 
     def readCSVFromString(s: String,
                           delimiter: Char = RecordSplitter.defaultFieldSeparator,
                           trimming: Trimming = NoAction,
-                          headers: Headers = ParseHeaders)(implicit typeTag: TypeTag[A]): List[Try[A]] = {
+                          headers: Headers = ParseHeaders,
+                          headerMapping: Map[String, String] = Map())(implicit typeTag: TypeTag[A]): List[Try[A]] = {
```

Comment:
thanks, lgtm - one minor nit, could use use Map.empty instead of Map()?

Reply:
Done.